grapher/axis/AxisViews.tsx

```diff
@@ -215,34 +248,26 @@ export class HorizontalAxisComponent extends React.Component<{
                         bounds.bottom - label.height
                     )}
                 {tickMarks}
-                {ticks.map((tick, i) => {
-                    const label = axis.formatTick(tick, {
-                        isFirstOrLastTick: i === 0 || i === ticks.length - 1,
-                    })
-                    const rawXPosition = axis.place(tick)
-                    // Ensure the first label does not exceed the chart viewing area
-                    const xPosition =
-                        i === 0
-                            ? Bounds.getRightShiftForMiddleAlignedTextIfNeeded(
-                                  label,
-                                  axis.tickFontSize,
-                                  rawXPosition
-                              ) + rawXPosition
-                            : rawXPosition
-                    const element = (
+                {tickLabels.map((label, i) => {
+                    const { x, xAlign, formattedValue } = label
+                    return (
                         <text
                             key={i}
-                            x={xPosition}
+                            // We use placeTickLabel() instead of using tick.x directly because the
+                            // range may have shifted e.g. to create space for the Y axis.
+                            // We use placeTickLabel() instead of place() because it sets an X
+                            // coordinate that may be different from the X coordinate of the tick mark.
```

Comment:
Apparently this is important because you had to say the same thing twice? And it seems like this isn't true anymore anyway?

Reply:
😬 Code comments are like

![code-comments](https://user-images.githubusercontent.com/1308115/127877136-6b104aee-1048-44bf-b118-d67811a04def.jpeg)